packages/web/src/store/token-dashboard/connectNewWalletSaga.ts

```diff
@@ -12,40 +12,57 @@ import { getWalletAddress } from './getWalletAddress'
 import { signMessage } from './signMessage'
 const { connectNewWallet } = tokenDashboardPageActions
 
-const { setIsConnectingWallet } = tokenDashboardPageActions
+const { setIsConnectingWallet, setModalState, resetStatus } =
+  tokenDashboardPageActions
 
 function* handleConnectNewWallet() {
-  const connection = yield* establishWalletConnection()
-  if (!connection) return
+  try {
+    const connection = yield* establishWalletConnection()
+    if (!connection) return
 
-  const { chain } = connection
+    const { chain } = connection
 
-  const walletAddress = yield* getWalletAddress(connection)
-  if (!walletAddress) return
+    const walletAddress = yield* getWalletAddress(connection)
+    if (!walletAddress) return
 
-  const isNewWallet = yield* checkIsNewWallet(walletAddress, chain)
-  if (!isNewWallet) return
+    const isNewWallet = yield* checkIsNewWallet(walletAddress, chain)
+    if (!isNewWallet) return
 
-  const { balance, collectibleCount } = yield* getWalletInfo(
-    walletAddress,
-    chain
-  )
+    const { balance, collectibleCount } = yield* getWalletInfo(
+      walletAddress,
+      chain
+    )
 
-  yield* put(
-    setIsConnectingWallet({
-      wallet: walletAddress,
-      chain,
-      balance,
-      collectibleCount
-    })
-  )
+    yield* put(
+      setIsConnectingWallet({
+        wallet: walletAddress,
+        chain,
+        balance,
+        collectibleCount
+      })
+    )
 
-  const signature = yield* signMessage(connection)
-  const updatedUserMetadata = yield* associateNewWallet(signature)
+    const signature = yield* signMessage(connection)
+    const updatedUserMetadata = yield* associateNewWallet(signature)
 
-  const disconnect = () => disconnectWallet(connection)
+    const disconnect = () => disconnectWallet(connection)
 
-  yield* addWalletToUser(updatedUserMetadata, disconnect)
+    yield* addWalletToUser(updatedUserMetadata, disconnect)
+  } catch (e) {
+    // Very likely we hit error path here i.e. user closes the web3 popup. Log it and restart
+    console.error(
```

Comment:
Maybe explicitly call sentry for better looking errors?

Reply:
mmm... now that I think about it, not sure these should go to sentry at all TBH. let me add some sort of regex

Reply:
not easily regexable since diff providers have diff errors. will send to sentry